packages/core/src/auth/utils.ts

```diff
@@ -93,12 +94,22 @@ export async function promptForConnection(auth: Auth, type?: 'iam' | 'iam-only'
 
 export async function promptAndUseConnection(...[auth, type]: Parameters<typeof promptForConnection>) {
```

Comment:
The PR description is a beautifully succinct outline of the "spec". Would this be the best place to add that "spec" as a docstring?

Reply:
added to addScopes instead, it is a common point for the prompter in toolkit and the login pages